applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_scan_for_conn_internal.h

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#ifndef _BT_MGMT_SCAN_FOR_CONN_H_
+#define _BT_MGMT_SCAN_FOR_CONN_H_
+
+#include <zephyr/bluetooth/bluetooth.h>
+
+/**
+ * @brief	Scan for a connection with a given device name
+ *
+ * @param	scan_param	The scan parameters to use (interval and window)
+ * @param	name		Device name to search for
+ *
+ * @return	0 if success, error otherwise
+ */
+int bt_mgmt_scan_for_conn_start(struct bt_le_scan_param *scan_param, char const *const name);
```

Comment:
I will change to const here as well

Reply:
What's more important is that you are storing the point of `name` in the scan manager; i.e. it is not enough to just make it `const const`, but it should in fact be `static`ally defined. If someone calls `bt_mgmt_scan_for_conn_start` with a stack-allocated `const const` name, then it will still fail miserably.

Reply:
bt_mgmt_scan_for_conn_start should only be called from bt_mgmt_scan_start though, that's why the header files are marked as _internal. I didn't want to store the name multiple places